lib/tpm2_hierarchy.c

```diff
@@ -81,32 +157,8 @@ bool tpm2_hierarchy_from_optarg(const char *value,
      * If the caller specifies the expected valid hierarchies, either as string,
      * or hex handles, they are additionally filtered here.
      */
-    if (!(flags & TPM2_HIERARCHY_FLAGS_O) && *hierarchy == TPM2_RH_OWNER) {
```

Comment:
aren't the NV and HIERARCHY flag ranges now mutually exclusive, you should probably check that. With that said, we can assume the caller knows what they are doing since it's an internal routine.